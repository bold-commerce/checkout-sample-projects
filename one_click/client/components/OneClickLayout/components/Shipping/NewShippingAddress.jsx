import React, { useState } from 'react';
import { useCountryInfo, useShippingAddress } from '@boldcommerce/checkout-react-components';
import { RadioItem } from '../RadioItem';
import { Address } from '../Address';

const NewShippingAddress = ({ selected, onChange, disabled, defaultAddress }) => {
  const {
    shippingAddress, countryInfo, shippingAddressErrors, submitShippingAddress,
  } = useShippingAddress();
  const [address, setAddress] = useState(shippingAddress);
  const {
    countries,
    provinces,
    showProvince,
    showPostalCode,
    provinceLabel,
  } = useCountryInfo(countryInfo, address);

  let provincePlaceholder = provinceLabel;

  // TODO: replace with languages config file
  if (provinceLabel === 'state_territory') {
    provincePlaceholder = 'state/territory';
  }

  return (
    <li>
      <RadioItem
        name="shipping_address"
        id={`address_0`}
        checked={selected}
        onChange={() => onChange({
          ...defaultAddress,
          id: null,
        })}
        disabled={disabled}
      >
        <p>Use new address</p>
      </RadioItem>
      {
        selected && (
          <div className="RadioButton RadioButton__NewAddressContainer">
            <Address
              address={address}
              onChange={(data) => setAddress((prevAddress) => ({
                ...prevAddress,
                ...data,
              }))}
              errors={shippingAddressErrors}
              countries={countries}
              provinces={provinces}
              showPostalCode={showPostalCode}
              showProvince={showProvince}
              submit={() => submitShippingAddress(address)}
              provinceLabel={provincePlaceholder}
            />
        </div>
        )
      }
    </li>
  )
};

export default NewShippingAddress;
