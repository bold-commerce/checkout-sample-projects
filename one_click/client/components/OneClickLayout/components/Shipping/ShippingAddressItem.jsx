import React from 'react';
import { RadioItem } from '../RadioItem';

const ShippingAddressItem = ({ address, onChange, selectedAddress, disabled }) => {
  const { address1, address2, city, province, zip, country, phone, first_name, last_name } = address;
  const addressArray = [address1, address2, city, province, zip, country];
  const addressCombined = addressArray.filter((addressString) => addressString !== '').join(', ');

  return (
    <li>
      <RadioItem 
        name="shipping_address"
        id={`address_${address.id}`}
        checked={address.id === selectedAddress}
        onChange={() => onChange(address)}
        disabled={disabled}
      >
        <p>{`${first_name} ${last_name}`}</p>
        <p>{addressCombined}</p>
        <p>{phone}</p>
      </RadioItem>
    </li>
  )
};

export default ShippingAddressItem;
