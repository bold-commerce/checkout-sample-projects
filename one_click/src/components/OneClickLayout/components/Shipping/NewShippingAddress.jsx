import React, { useState, useCallback } from 'react';
import { useCountryInfo, useShippingAddress } from '@boldcommerce/checkout-react-components';
import { RadioItem } from '../RadioItem';
import { Address } from '../Address';
import { useTranslation } from 'react-i18next';

const NewShippingAddress = ({ selected, onChange, disabled, defaultAddress }) => {
  const { data: shippingAddress, errors, submitShippingAddress } = useShippingAddress();
  const [address, setAddress] = useState(shippingAddress);
  const { data: countryInfo } = useCountryInfo(address);
  const { t } = useTranslation();

  let provincePlaceholder = countryInfo.provinceLabel;

  const handleOnChange = useCallback((currentAddress) => {
    setAddress(currentAddress);
    onChange(currentAddress);
  }, [])

  return (
    <li>
      <RadioItem
        name="shipping_address"
        id={`address_0`}
        checked={selected}
        onChange={() => handleOnChange({
          ...defaultAddress,
          id: null,
        })}
        disabled={disabled}
      >
        <p>{t('shipping.use_new_address')}</p>
      </RadioItem>
      {
        selected && (
          <div className="RadioButton RadioButton__NewAddressContainer">
            <Address
              address={address}
              onChange={(d) => setAddress((prevAddress) => ({
                ...prevAddress,
                ...d,
              }))}
              errors={errors}
              countries={countryInfo.countries}
              provinces={countryInfo.provinces}
              showPostalCode={countryInfo.showPostalCode}
              showProvince={countryInfo.showProvince}
              submit={() => submitShippingAddress(address)}
              provinceLabel={provincePlaceholder}
            />
        </div>
        )
      }
    </li>
  )
};

export default NewShippingAddress;
