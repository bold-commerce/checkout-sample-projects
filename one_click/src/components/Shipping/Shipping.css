ul.AddressList {
  list-style: none;
  display: grid;
  grid-auto-flow: row;
}

li .RadioItem {
  display: flex;
  column-gap: 8px;
  border-bottom: 1px solid rgba(0, 0, 0, 0.4);
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  color: rgba(0, 0, 0, 0.65);
  padding: 8px 0;
}

li .RadioItem input {
  margin-top: 4px;
}

.RadioItem__Content > p:first-child {
  color: rgba(0, 0, 0, 0.9);
}

.RadioButton.RadioButton__NewAddressContainer {
  padding: 8px 0;
}

.Shipping__ShippingAddress {
  margin: 16px 0;
}
