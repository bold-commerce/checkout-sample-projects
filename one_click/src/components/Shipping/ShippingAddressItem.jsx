import React from 'react';
import { RadioItem } from '../RadioItem';

const ShippingAddressItem = ({ address, onChange, selectedAddress, disabled }) => {
  const { address_line_1, address_line_2, city, province, postal_code, country, phone_number, first_name, last_name } = address;
  const addressArray = [address_line_1, address_line_2, city, province, postal_code, country];
  const addressCombined = addressArray.filter((addressString) => addressString !== '').join(', ');

  return (
    <li>
      <RadioItem 
        name="shipping_address"
        id={`address_${address.id}`}
        checked={address.id === selectedAddress}
        onChange={() => onChange(address)}
        disabled={disabled}
      >
        <p>{`${first_name} ${last_name}`}</p>
        <p>{addressCombined}</p>
        <p>{phone_number}</p>
      </RadioItem>
    </li>
  )
};

export default ShippingAddressItem;
