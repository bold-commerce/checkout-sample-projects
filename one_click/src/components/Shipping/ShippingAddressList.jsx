import React, { useEffect, useMemo, useState } from 'react';
import NewShippingAddress from './NewShippingAddress';
import ShippingAddressItem from './ShippingAddressItem';

const ShippingAddressList = ({ addresses, onChange, selectedAddress, disabled }) => {
  const [prevAddress, setPrevAddress] = useState(null);

  const shippingAddressList = addresses.map((address) => (
    <ShippingAddressItem
      address={address}
      key={address.id}
      onChange={onChange}
      selectedAddress={selectedAddress}
      disabled={disabled}
    />
  ));

  useEffect(() => {
    if (selectedAddress) {
      setPrevAddress(selectedAddress);
    }
  }, [selectedAddress]);

  const lastSelectedAddress = useMemo(() => {
    if (prevAddress) {
      return addresses.find((address) => address.id === prevAddress);
    } else {
      return addresses[0];
    }
  }, [prevAddress])

  // TODO: On select of new address, remove old address from order
  return (
    <ul className="AddressList">
      {shippingAddressList}
      <NewShippingAddress
        selected={selectedAddress === null || !selectedAddress}
        onChange={onChange}
        disabled={disabled}
        defaultAddress={lastSelectedAddress}
      />
    </ul>
  )
};

export default ShippingAddressList;
