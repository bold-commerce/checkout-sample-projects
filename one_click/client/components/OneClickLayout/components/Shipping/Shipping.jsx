import React, { useCallback } from 'react';
import { useLoadingStatus, useSavedAddresses, useShippingAddress } from '@boldcommerce/checkout-react-components';
import { BackButton } from '../BackButton';
import ShippingAddressList from './ShippingAddressList';
import ShippingLines from '../ShippingLines/ShippingLines';
import './Shipping.css';
import { Header } from '../Header';

const Shipping = () => {
  const { data: shippingAddress, submitShippingAddress } = useShippingAddress();
  const { data: savedAddresses } = useSavedAddresses();
  const { shippingAddressLoadingStatus, shippingLinesLoadingStatus } = useLoadingStatus();
  const handleSubmit = useCallback(async (address) => {
    try {
      await submitShippingAddress(address);
    }
    catch(e) {
    }
  }, []);

  const isSetting = shippingAddressLoadingStatus === 'setting' || shippingLinesLoadingStatus === 'fetching';

  return (
    <div className="Shipping">
      <Header title={"Shipping"}/>
      <BackButton />
      <section className="Shipping__ShippingAddress">
        <h3>Shipping address</h3>
        <ShippingAddressList
          addresses={savedAddresses}
          onChange={(address) => handleSubmit(address)}
          selectedAddress={shippingAddress?.id}
          disabled={isSetting}
        />
      </section>
      <ShippingLines />
    </div>
  );
};

export default Shipping;
