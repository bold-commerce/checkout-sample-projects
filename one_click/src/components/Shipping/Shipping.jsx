import React, { useCallback, useState } from 'react';
import { useLoadingStatus, useSavedAddresses, useShippingAddress } from '@boldcommerce/checkout-react-components';
import { BackButton } from '../BackButton';
import ShippingAddressList from './ShippingAddressList';
import ShippingLines from '../ShippingLines/ShippingLines';
import { Header } from '../Header';
import classNames from 'classnames';
import { useAnalytics, useErrorLogging } from '../../hooks';
import './Shipping.css';
import { useTranslation } from 'react-i18next';

const Shipping = ({ show, onBack }, ref) => {
  const { data: shippingAddress, errors, submitShippingAddress } = useShippingAddress();
  const { data: savedAddresses } = useSavedAddresses();
  const { shippingAddressLoadingStatus, shippingLinesLoadingStatus } = useLoadingStatus();
  const [selectingAddress, setSelectingAddress] = useState(false);
  const trackEvent = useAnalytics();
  const logError = useErrorLogging();
  const { t } = useTranslation();

  const handleSubmit = useCallback(async (address) => {
    setSelectingAddress(true);
    try {
      await submitShippingAddress(address);
      trackEvent('set_shipping_address');
    }
    catch(e) {
      logError('shipping_address', e);
    }
    setSelectingAddress(false);
  }, []);

  const isSetting = shippingAddressLoadingStatus === 'setting' || shippingLinesLoadingStatus === 'fetching' || selectingAddress;

  return (
    <div ref={ref} className={classNames('Sidebar Shipping', show ? 'Sidebar--Show' : 'Sidebar--Hide')}>
      <Header title={"Shipping"} />
      <BackButton onClick={onBack} />
      <section className="Shipping__ShippingAddress">
        <h3 className="FieldSet__Heading">{t('shipping.address')}</h3>
        <ShippingAddressList
          addresses={savedAddresses}
          onChange={(address) => handleSubmit(address)}
          selectedAddress={shippingAddress?.id}
          disabled={isSetting}
        />
      </section>
      <ShippingLines activePage={show} disabled={Boolean(errors)} />
    </div>
  );
};

const ShippingForwardedRef = React.forwardRef(Shipping);

export default ShippingForwardedRef;
