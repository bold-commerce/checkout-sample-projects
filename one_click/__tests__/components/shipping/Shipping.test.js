import React from 'react';
import Shipping from '../../../src/components/OneClickLayout/components/Shipping/Shipping';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import { exampleShippingState as MOCKexampleShippingState } from '../../utils/shippingLinesHelper';
import {
    countries as MOCKcountries,
    caProvinces as MOCKprovinces,
    exampleAddress as MOCKexampleAddress,
    exampleSavedAddresses as MOCKexampleSavedAddresses
 } from '../../utils/addressHelpers';

 jest.mock('@boldcommerce/checkout-react-components', () => ({
     ...jest.requireActual('@boldcommerce/checkout-react-components'),
     useLoadingStatus: () => ({
        customerLoadingStatus: "fulfilled",
        discountLoadingStatus: "fulfilled",
        isLeading: true,
        lineItemsLoadingStatus: "fulfilled",
        shippingAddressLoadingStatus: "fulfilled",
        shippingLinesLoadingStatus: "fulfilled"
    }),
    useShippingLines: () => ({
        data: MOCKexampleShippingState,
        loadindStatus: 'fulfilled',
        errors: [],
        updateShippingLines: (() => {})
    }),
    useCountryInfo: () => ({
        data: {
            countries: MOCKcountries,
            provinceLabel: "province",
            provinces: MOCKprovinces,
            showPostalCode: true,
            showProvince: true
        }
    }),
    useSavedAddresses: () => ({
        data: MOCKexampleSavedAddresses
    }),
    useShippingAddress: () => ({
        data: MOCKexampleAddress,
        submitShippingAdderess: (() => {})
    })
})).mock('react', () => ({
    ...jest.requireActual('react'),
    useContext: () => ({
        setShowCheckout: (() => {})
    })
}));

describe('Shipping', () => {
    test('renders Shipping component', () => {
        const { asFragment } = render(
            <Shipping />,
            {wrapper: MemoryRouter}
        );
        expect(asFragment()).toMatchSnapshot();
    });
});