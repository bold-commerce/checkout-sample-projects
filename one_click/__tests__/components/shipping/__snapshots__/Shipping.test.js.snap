// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shipping renders Shipping component 1`] = `
<DocumentFragment>
  <div
    class="Sidebar Shipping Sidebar--Hide"
  >
    <div
      class="Header"
    >
      <h1
        class="Header__Title"
      >
        Shipping
      </h1>
      <button
        class="Header__Times"
      >
        <svg
          fill="none"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            opacity="0.8"
          >
            <path
              clip-rule="evenodd"
              d="M17.7071 7.70706L7.70712 17.7071L6.29291 16.2928L16.2929 6.29285L17.7071 7.70706Z"
              fill="black"
              fill-rule="evenodd"
            />
            <path
              clip-rule="evenodd"
              d="M16.2929 17.7071L6.29291 7.70706L7.70712 6.29285L17.7071 16.2928L16.2929 17.7071Z"
              fill="black"
              fill-rule="evenodd"
            />
          </g>
        </svg>
      </button>
    </div>
    <div>
      <button
        class="back-btn"
        type="button"
      >
        Back
      </button>
    </div>
    <section
      class="Shipping__ShippingAddress"
    >
      <h3
        class="FieldSet__Heading"
      >
        Shipping address
      </h3>
      <ul
        class="AddressList"
      >
        <li>
          <div
            class="RadioItem"
          >
            <input
              id="address_1"
              name="shipping_address"
              type="radio"
            />
            <label
              class="RadioItem__Content"
              for="address_1"
            >
              <p>
                John Doe
              </p>
              <p>
                123 Fake Street, Unit 12, Winnipeg, Manitoba, R3A1B3, Canada
              </p>
              <p>
                204-123-4567
              </p>
            </label>
          </div>
        </li>
        <li>
          <div
            class="RadioItem"
          >
            <input
              id="address_2"
              name="shipping_address"
              type="radio"
            />
            <label
              class="RadioItem__Content"
              for="address_2"
            >
              <p>
                John Doe
              </p>
              <p>
                456 Test Drive, Winnipeg, Manitoba, R3A1B3, Canada
              </p>
              <p>
                204-123-4567
              </p>
            </label>
          </div>
        </li>
        <li>
          <div
            class="RadioItem"
          >
            <input
              id="address_3"
              name="shipping_address"
              type="radio"
            />
            <label
              class="RadioItem__Content"
              for="address_3"
            >
              <p>
                John Doe
              </p>
              <p>
                789 Fake Crescent, Winnipeg, Manitoba, R3A1B3, Canada
              </p>
              <p>
                204-123-4567
              </p>
            </label>
          </div>
        </li>
        <li>
          <div
            class="RadioItem"
          >
            <input
              checked=""
              id="address_0"
              name="shipping_address"
              type="radio"
            />
            <label
              class="RadioItem__Content"
              for="address_0"
            >
              <p>
                Use new address
              </p>
            </label>
          </div>
          <div
            class="RadioButton RadioButton__NewAddressContainer"
          >
            <div
              class="FieldSet--Address"
            >
              <div
                class="FieldGroup"
              >
                <div
                  class="stx-field stx-field--with-input Field Field--FirstName"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      class="stx-input__input-element"
                      id="stxField1"
                      name="first_name"
                      placeholder="First name (optional)"
                      type="text"
                      value="John"
                    />
                  </div>
                </div>
                <div
                  class="stx-field stx-field--with-input Field Field--LastName"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      class="stx-input__input-element"
                      id="stxField2"
                      name="last_name"
                      placeholder="Last name (optional)"
                      type="text"
                      value="Doe"
                    />
                  </div>
                </div>
              </div>
              <div
                class="FieldGroup"
              >
                <div
                  class="stx-field stx-field--with-input Field Field--Company"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      class="stx-input__input-element"
                      id="stxField3"
                      name="business_name"
                      placeholder="Company (optional)"
                      type="text"
                      value="Bold Commerce"
                    />
                  </div>
                </div>
              </div>
              <div
                class="FieldGroup"
              >
                <div
                  class="stx-field stx-field--with-input Field Field--Address"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      class="stx-input__input-element"
                      id="stxField4"
                      name="address_line_1"
                      placeholder="Address (optional)"
                      type="text"
                      value="123 Fake Street"
                    />
                  </div>
                </div>
                <div
                  class="stx-field stx-field--with-input Field Field--Address2"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      class="stx-input__input-element"
                      id="stxField5"
                      name="address_line_2"
                      placeholder="Apt, suite, etc."
                      type="text"
                      value="Unit 12"
                    />
                  </div>
                </div>
              </div>
              <div
                class="FieldGroup"
              >
                <div
                  class="stx-field stx-field--with-input Field Field--City"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      class="stx-input__input-element"
                      id="stxField6"
                      name="city"
                      placeholder="City (optional)"
                      type="text"
                      value="Winnipeg"
                    />
                  </div>
                </div>
              </div>
              <div
                class="FieldGroup FieldGroup__Country"
              >
                <div
                  class="stx-field stx-field--with-select SelectField Field--Country"
                >
                  <div
                    class="stx-select stx-select--with-placeholder"
                  >
                    <select
                      class="stx-select__select-element"
                      id="stxField7"
                    >
                      <option
                        hidden=""
                        value="___PLACEHOLDER___"
                      >
                        Select a country
                      </option>
                      <option
                        value="CA"
                      >
                        Canada
                      </option>
                      <option
                        value="US"
                      >
                        United States
                      </option>
                    </select>
                    <span
                      class="stx-select__icon"
                    >
                      <svg
                        class="stx-select-icon__icon"
                        viewBox="0 0 24 24"
                      >
                        <path
                          class="stx-select-icon__path"
                          d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                          fill="#6B6B6B"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                <div
                  class="stx-field stx-field--with-select SelectField Field--Province"
                >
                  <div
                    class="stx-select stx-select--with-placeholder"
                  >
                    <select
                      class="stx-select__select-element"
                      id="stxField8"
                    >
                      <option
                        hidden=""
                        value="___PLACEHOLDER___"
                      >
                        Select a province
                      </option>
                      <option
                        value="AB"
                      >
                        Alberta
                      </option>
                      <option
                        value="BC"
                      >
                        British Columbia
                      </option>
                      <option
                        value="MB"
                      >
                        Manitoba
                      </option>
                      <option
                        value="NB"
                      >
                        New Brunswick
                      </option>
                      <option
                        value="NL"
                      >
                        Newfoundland and Labrador
                      </option>
                      <option
                        value="NT"
                      >
                        Northwest Territories
                      </option>
                      <option
                        value="NS"
                      >
                        Nova Scotia
                      </option>
                      <option
                        value="NU"
                      >
                        Nunavut
                      </option>
                      <option
                        value="ON"
                      >
                        Ontario
                      </option>
                      <option
                        value="PE"
                      >
                        Prince Edward Island
                      </option>
                      <option
                        value="QC"
                      >
                        Quebec
                      </option>
                      <option
                        value="SK"
                      >
                        Saskatchewan
                      </option>
                      <option
                        value="YT"
                      >
                        Yukon Territory
                      </option>
                    </select>
                    <span
                      class="stx-select__icon"
                    >
                      <svg
                        class="stx-select-icon__icon"
                        viewBox="0 0 24 24"
                      >
                        <path
                          class="stx-select-icon__path"
                          d="M3.75261265,6.34153452 C3.38894463,5.92588561 2.75718344,5.88374677 2.34153452,6.24741478 C1.92588561,6.6110828 1.88374677,7.24284399 2.24741478,7.65849291 L11.2524148,18.2364929 C11.6508167,18.6918404 12.3591606,18.6918563 12.757583,18.2365268 L21.763583,7.6585268 C22.1272697,7.24289426 22.0851593,6.61113117 21.6695268,6.24744443 C21.2538943,5.8837577 20.6221312,5.9258681 20.2584444,6.34150064 L12.0050479,16.0594019 L3.75261265,6.34153452 Z"
                          fill="#6B6B6B"
                        />
                      </svg>
                    </span>
                  </div>
                </div>
                <div
                  class="stx-field stx-field--with-input Field Field--Postal_Code"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      class="stx-input__input-element"
                      id="stxField9"
                      name="postal"
                      placeholder="Postal code / ZIP"
                      type="text"
                      value="R3A1B3"
                    />
                  </div>
                </div>
              </div>
              <div
                class="FieldGroup"
              >
                <div
                  class="stx-field stx-field--with-input Field Field--Phone"
                >
                  <div
                    class="stx-input"
                  >
                    <input
                      class="stx-input__input-element"
                      id="stxField10"
                      name="phone_number"
                      placeholder="Phone (optional)"
                      type="tel"
                      value="204-123-4567"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </section>
    <section
      class="FieldSet FieldSet--ShippingMethod"
    >
      <div
        class="FieldSet__Header"
      >
        <h3
          class="FieldSet__Heading"
        >
          Shipping method
        </h3>
      </div>
      <div
        class="FieldSet__Content"
      >
        <div
          class="FieldSet--EmptyState"
        >
          <svg
            fill="none"
            height="29"
            viewBox="0 0 28 29"
            width="28"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="#C4C4C4"
              height="5"
              rx="2.5"
              width="28"
            />
            <rect
              fill="#C4C4C4"
              height="5"
              rx="2.5"
              width="28"
              y="8"
            />
            <rect
              fill="#C4C4C4"
              height="5"
              rx="2.5"
              width="28"
              y="16"
            />
            <rect
              fill="#C4C4C4"
              height="5"
              rx="2.5"
              width="28"
              y="24"
            />
            <circle
              cx="2.5"
              cy="10.5"
              fill="white"
              r="1.5"
            />
            <circle
              cx="2.5"
              cy="2.5"
              fill="white"
              r="1.5"
            />
            <circle
              cx="2.5"
              cy="18.5"
              fill="white"
              r="1.5"
            />
            <circle
              cx="2.5"
              cy="26.5"
              fill="white"
              r="1.5"
            />
          </svg>
          To view shipping options, complete filling in your address
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
